srcts/src/shiny/init.ts

```diff
@@ -150,6 +150,22 @@ function initShiny(windowShiny: Shiny): void {
     (x) => x.value
   );
 
+  // If and when input/output registration happens in the future,
+  // check to see if renderHtml()/renderContent() is currently executing.
+  // If it isn't, we likely need to bind again to the entire document,
+  // since Shiny has bound to the DOM before having access to the full
+  // set of bindings. (#3635)
+  const debouncedBindAll = debounce(0, () =>
+    windowShiny.bindAll?.(document.documentElement)
+  );
+
+  const maybeBindOnRegister = () => {
+    if (!renderHtml.isExecuting()) debouncedBindAll();
+  };
+
+  inputBindings.onRegister(maybeBindOnRegister, false);
+  outputBindings.onRegister(maybeBindOnRegister, false);
```

Comment:
A rough suggestion:

```suggestion
  let enqueuedCount = 0;
  const enqueueRebind = () => {
    enqueuedCount++;

    ShinyApp.actionQueue.enqueue(() => {
      enqueuedCount--;
      // If this function is scheduled more than once in the queue, don't do anything.
      // Only do the bindAll when we get to the last instance of this function in the queue.
      if (enqueuedCount === 0) {
        windowShiny.bindAll?.(document.documentElement)
      }
    });
  };

  inputBindings.onRegister(enqueueRebind, false);
  outputBindings.onRegister(enqueueRebind, false);
```

Notes:
* Expand on the comment, otherwise we'll forget what this is all about.
* Check to make sure this callback doesn't fire during init.